controls/V-73011.rb

```diff
@@ -1,77 +1,92 @@
+  functionality exceeding requirements or mission objectives.  
+
+  PostgreSQL must adhere to the principles of least functionality by
+  providing only essential capabilities. 
+
+  Unused, unnecessary PostgreSQL components increase the attack vector for
+  PostgreSQL by introducing additional targets for attack. By minimizing the
+  services and applications installed on the system, the number of potential
+  vulnerabilities is reduced. Components of the system that are unused and cannot
+  be uninstalled must be disabled. The techniques available for disabling
+  components will vary by DBMS product, OS and the nature of the component and
+  may include DBMS configuration settings, OS service settings, OS file access
+  security, and DBMS user/role permissions."
+
   impact 0.5
   tag "severity": "medium"
   tag "gtitle": "SRG-APP-000141-DB-000092"
   tag "gid": "V-73011"
-  tag "rid": "SV-87663r1_rule"
+  tag "rid": "SV-87663r2_rule"
   tag "stig_id": "PGS9-00-009200"
+  tag "fix_id": "F-79457r1_fix"
   tag "cci": ["CCI-000381"]
   tag "nist": ["CM-7 a", "Rev_4"]
-  tag "check": "To list all installed packages, as the system administrator, run the
-following:
-
-# RHEL/CENT Systems
-$ sudo yum list installed | grep postgres
-
-# Debian Systems
-$ dpkg --get-selections | grep postgres
+  tag "false_negatives": nil
+  tag "false_positives": nil
+  tag "documentable": false
+  tag "mitigations": nil
+  tag "severity_override_guidance": false
+  tag "potential_impacts": nil
+  tag "third_party_tools": nil
+  tag "mitigation_controls": nil
+  tag "responsibility": nil
+  tag "ia_controls": nil
+  desc "check", "To list all installed packages, as the system administrator, run
+  the following:
 
-If any packages are installed that are not required, this is a finding."
-  tag "fix": "To remove any unneeded executables, as the system administrator, run
-the following:
+  # RHEL/CENT Systems
+  $ sudo yum list installed | grep postgres
 
-# RHEL/CENT Systems
-$ sudo yum erase <package_name>
+  # Debian Systems
+  $ dpkg --get-selections | grep postgres
 
-# Debian Systems
-$ sudo apt-get remove <package_name>"
+  If any packages are installed that are not required, this is a finding."
+  
+  desc "fix", "To remove any unneeded executables, as the system administrator,
+  run the following:
 
-# @todo how do I identify the packages that are not required for the current OS? need datafile of approved?
-# @todo assume need two tests, one for RHEL/CENT, and one for Debian?
+  # RHEL/CENT Systems
+  $ sudo yum erase <package_name>
 
-  only_if { false }
+  # Debian Systems
+  $ sudo apt-get remove <package_name>"
 
-end
+  if command('yum').exist?
+    lines = command("yum list installed | grep \"postgres\"")
+    lines.each do |package|
+      describe(package) do
+        it { should be_in approved_packages }
+      end
+    end
+  end
+  
+  if command('dpkg').exist?
+    lines = command("yum list installed | grep \"postgres\"")
```

Comment:
Change 'lines' to 'dpkg_packages' to be more descriptive.

The command should be the following for Debian:
dpkg --get-selections | grep postgres